compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/AIETarget.cpp

```diff
@@ -130,813 +135,118 @@ class AIETargetBackend final : public IREE::HAL::TargetBackend {
+    if (err) {
+      return moduleOp.emitOpError() << "failed to create temporary working "
+                                       "directory for xclbin generation: "
+                                    << err.message();
+    }
   }
 
-  std::vector<uint32_t> ipuInstrs;
-  if (failed(generateIPUInstructions(context, moduleOp, ipuInstrs)))
-    return failure();
-
-  // dump lx6 instructions, if required.
-  if (!serOptions.dumpIntermediatesPath.empty()) {
-    std::string dumpString;
-    llvm::raw_string_ostream dumpStream(dumpString);
-    for (auto w : ipuInstrs) {
-      dumpStream << llvm::format("%08X\n", w);
+  std::string errorMessage;
+  SmallString<128> inputMlirPath(workDir);
+  auto basename =
+      llvm::join_items("_", serOptions.dumpBaseName, variantOp.getName());
+  llvm::sys::path::append(inputMlirPath, basename + ".aiecc.mlir");
+  {
+    auto inputMlirOut = openOutputFile(inputMlirPath, &errorMessage);
+    if (!inputMlirOut) {
+      return moduleOp.emitOpError() << "Failed to write MLIR: " << errorMessage;
     }
-    IREE::HAL::dumpDataToPath(serOptions.dumpIntermediatesPath,
-                              serOptions.dumpBaseName, variantOp.getName(),
-                              ".insts.txt",
-                              StringRef(dumpString.data(), dumpString.size()));
+    moduleOp.print(inputMlirOut->os(), OpPrintingFlags().useLocalScope());
+    inputMlirOut->keep();
   }
 
-  XclBinGeneratorKit toolkit(options.peanoInstallDir, options.vitisInstallDir,
-                             options.showInvokedCommands);
+  SmallString<128> aie2xclbin(options.mlirAieInstallDir);
+  llvm::sys::path::append(aie2xclbin, "bin", "aie2xclbin");
+  SmallString<128> XCLBinPath(workDir);
+  llvm::sys::path::append(XCLBinPath, basename + ".xclbin");
+  SmallString<128> IPUInstPath(workDir);
+  llvm::sys::path::append(IPUInstPath, basename + ".ipu.txt");
 
-  FailureOr<Artifact> objFile;
-  {
-    OpBuilder::InsertionGuard g(executableBuilder);
-    executableBuilder.setInsertionPoint(moduleOp);
-    auto clonedModuleOp =
-        dyn_cast<ModuleOp>(executableBuilder.clone(*moduleOp.getOperation()));
-    objFile = generateObjectFile(context, clonedModuleOp, getOptions(), toolkit,
-                                 serOptions.dumpIntermediatesPath,
-                                 serOptions.dumpBaseName);
-    if (failed(objFile)) {
-      return moduleOp.emitOpError("failed binary conversion using Peano");
-    }
-    clonedModuleOp->erase();
+  // collect names of kernels as they need to be in kernels.json
+  // generated by `aie2xclbin`
+  SmallVector<std::string> entryPointNames;
+  for (auto exportOp : variantOp.getExportOps()) {
+    entryPointNames.emplace_back(exportOp.getSymName());
   }
 
-  // Create a temporary work directory needed for subsequent commands.
-  FailureOr<std::string> workDir =
-      Artifact::createTemporaryDirectory(variantOp.getName());
-  if (failed(workDir)) {
+  if (entryPointNames.size() == 0) {
```

Comment:
Nit: You can just have `if (entryPointsNames.size()  == 1) { return ... "expected single entry point");}` for now.